Add docs for notifying transactions via plugin (#22097)
solana-labs/solana: docs/src/developing/plugins/accountsdb_plugin.md

the validator in case of error persisting to external stores. When the
validator restarts the account data will be re-transmitted.

The following method is used for notifying transactions:

```
    fn notify_transaction(
        &mut self,
        transaction: ReplicaTransactionInfoVersions,
        slot: u64,
    ) -> Result<()>
```

> The `ReplicaAccountInfReplicaTransactionInfoVersionsoVersions` struct

Reviewer: ```suggestion
The `ReplicaTransactionInfoVersionsoVersions` struct
```

> contains the information about a transaction streamed. It wraps `ReplicaTransactionInfo`

Reviewer: ```suggestion
contains the information about a streamed transaction. It wraps `ReplicaTransactionInfo`
```

    }
```

### Transaction Selection

`transaction_selector`, controls if and what transactions to store.

> If this field is missing none of the transctions is stored.

Reviewer: ```suggestion
If this field is missing, none of the transactions are stored.
```

For example, one can use the following to select only the transactions
referencing accounts with particular Base58-encoded Pubkeys,

```
"transaction_selector" : {
    "mentions" : \["pubkey-1", "pubkey-2", ..., "pubkey-n"\],
}
```

> The `mentions` field support wildcard to select all transaction or

Reviewer: ```suggestion
The `mentions` field supports wildcards to select all transaction or
```

            where ranked.rnk > 1000)
```

### Main Tables

The following are the tables in the Postgres database

| Table         | Description             |
|:--------------|:------------------------|

> | account       | Accounts data           |

Reviewer: ```suggestion
| account       | Account data            |
```
